### PR TITLE
Encryption failure test case fix

### PR DESCRIPTION
**Pull request by GPrabhudas**
address issue: https://github.com/apache/pulsar-client-go/issues/707

### CODE REVIEW

File: pulsar/producer_impl.go [hunk 1]
```diff
@@ -124,6 +126,25 @@ func newProducer(client *client, options *ProducerOptions) (*producer, error) {
 		}
 	}
 
+	encryption := options.Encryption
+	// add default message crypto if not provided
+	if encryption != nil && len(encryption.Keys) > 0 {
+		if encryption.KeyReader == nil {
```
Comment (wolfstudy) [inline pulsar/producer_impl.go:132]: It seems that there is no similar check in `newConsumer` and `newReader`, do they not need it?

**Review by wolfstudy — COMMENTED** [review]
@GPrabhudas Nice work, just a little comment, please check.

File: pulsar/producer_impl.go (hunk 1 above)
Comment (GPrabhudas) [reply pulsar/producer_impl.go:132]: Similar check is not needed on consumer/reader  side.

**Review by wolfstudy — APPROVED** [review]
LGTM +1

**Pull request merged**